app/src/main/java/org/simple/clinic/registration/register/RegistrationLoadingScreenController.kt

```diff
@@ -30,6 +30,7 @@ class RegistrationLoadingScreenController @Inject constructor(
 
     return Observable.merge(creates, retryClicks)
         .flatMapSingle { userSession.register() }
+        .doOnNext { if (it is Success) userSession.clearOngoingRegistrationEntry() }
```

Comment:
Since `UserSession#clearOngoingRegistrationEntry` returns a `Completable`, this call does not do anything because that returned `Completable` is never subscribed to.

@ragunathjawahar We have run into a few cases like this and it's dependent on the reviewer to catch these errors. We should discuss how we can add checks in place to catch problems like this without depending on humans.

Reply:
I agree.

Reply:
I have fixed this code for now by subscribing to this `Completable` explicitly.